openlibrary/macros/databarWork.html

```diff
@@ -10,15 +10,12 @@
 $ viewbook = "//%s/stream/XXX?ref=ol" % bookreader_host()
 $ prices = page.key.startswith('/books/')
 
-$ isbn = (page.isbn_10 and page.isbn_10[0]) or (page.isbn_13 and page.isbn_13[0]) or ""
-$ isbn = isbn.replace("-", "")
 $ oclc_numbers = (page.oclc_numbers and page.oclc_numbers[0]) or ""
-$ isbn_13 = (page.isbn_13 and page.isbn_13[0]) or ""
-$ isbn_10 = ""
-$if page.isbn_10:
-    $ isbn_10 = page.isbn_10[0].replace("-", "")
-$elif isbn_13:
-    $ isbn_10 = isbn_13_to_isbn_10(isbn_13)
+$ isbn_13 = (page.isbn_13 and page.isbn_13[0]).replace("-", "") or ""
+$ isbn_10 = (page.isbn_10 and page.isbn_10[0]).replace("-", "") or ""
```

Comment:
These will error if `page.isbn_13` is `None` I believe cause it'll call `None.replace`. What type is `page`? Can we add a class method `.getIsbn13()` that handles all these edge cases?

Reply:
This logic seems a bit confusing, and is near duplicated below in edition sort. It's not completely clear what this code is trying to do, it seems to be trying to pick a single ISBN out of all possible candidates? It's not clear whether the preferred form is ISBN 10 or 13, and the two pages handle obtaining an ASIN differently.   

This code (including the stripping of hyphens) looks like it should not be in every web template, but on the book model or similar so all the templates that need it can call `<object>.isbn()` and get the same best isbn, in the correct format, and `<object>.asin()` to get the best asin.

Reply:
@hornc Do those methods exist? Can we try adding a method to one of the `book` or `edition` models? I have a feeling we _do_ have access to those methods from the templates.

Reply:
done